#17: Layout of Stream Viewing Page

emilio-alvarez opened the pull request:
Layout the stream viewing page to be a column for the stream, and another column for the chat. When a description is long, a scrollbar will appear for the stream column. If there's a lot of messages, the chat will have a scrollbar. The message box stays fixed in place.

This PR also makes the app use the entire width and height of the page.

When content fits the page:
![image](https://user-images.githubusercontent.com/25234283/78201277-77899d80-744e-11ea-9102-0a0e01f1b47c.png)

When there's a long description or something, the scroll bar shows up for the viewport, but not the page:
![image](https://user-images.githubusercontent.com/25234283/78201224-51fc9400-744e-11ea-9c79-5111df2344cf.png)

Likewise, if there's a lot of messages, only the chat scrolls, and not the entire page.

A dummy chat component was added just to account for it in the layout. The chat can be improved while working on #9 .

Resolves #17

emilio-alvarez committed:
Ignore .vscode workspace settings.

emilio-alvarez committed:
Use sha1 for integrity of packages.

This was done by npm automatically as far as I can tell.

Not sure why this is being done, but oh well.

emilio-alvarez committed:
Basic Grid layout. StreamChat dummy component.

The stream chat should be expanded in the future.

Just added so I can work on the grid layout of the stream page.

emilio-alvarez committed:
Merge branch 'master' into 17-view-stream-layout

emilio-alvarez committed:
Merge branch 'master' into 17-view-stream-layout

emilio-alvarez committed:
Made content use 100% width.

emilio-alvarez committed:
Getting the profile of the user in StreamShow.

emilio-alvarez committed:
Fix invalid property. autocomplete -> autoComplete

emilio-alvarez committed:
Merge branch 'master' into 17-view-stream-layout

emilio-alvarez committed:
Fetch the user info after stream info.

Probably a better implementation than before.

Unfortunately inner joins aren't a thing in NRDB, so it has to be done
this way.

emilio-alvarez committed:
Added catch in case there is an error.

emilio-alvarez committed:
Stretching the app to fill screen. Stream header.

emilio-alvarez committed:
Stream is scrollable. App is full page.

emilio-alvarez committed:
Chat also uses full page.

emilio-alvarez committed:
Merge branch 'master' into 17-view-stream-layout

emilio-alvarez committed:
Moved some css around.

josh-velasquez commented:
> Layout the stream viewing page to be a column for the stream, and another column for the chat. When a description is long, a scrollbar will appear for the stream column. If there's a lot of messages, the chat will have a scrollbar. The message box stays fixed in place.
> 
> This PR also makes the app use the entire width and height of the page.
> 
> When content fits the page:
> ![image](https://user-images.githubusercontent.com/25234283/78188095-e48c3b80-742c-11ea-95d2-3cc6717a51ff.png)
> 
> When there's a long description or something, the scroll bar shows up for the viewport, but not the page:
> ![image](https://user-images.githubusercontent.com/25234283/78188133-f2da5780-742c-11ea-9886-36659f48f85e.png)
> Likewise, if there's a lot of messages, only the chat scrolls, and not the entire page.
> 
> A dummy chat component was added just to account for it in the layout. The chat can be improved while working on #9 .
> 
> Resolves #17

is it possible to add some padding at the bottom? When loading the page it seems cutoff when at full height (at the bottom at least)
![Capture](https://user-images.githubusercontent.com/32565196/78191302-f670dd00-7432-11ea-90a4-492a0cfe875a.PNG)

emilio-alvarez committed:
Last child of scrollable has margin. Chat padding.

The chat had its padding re-added so that it's the default that semantic
gives it.

Added a style that adds a small margin to the bottom of the last child
of a scrollable element. This makes it so that the content doesn't look
cut off, and the user knows they've reached the bottom.

josh-velasquez reviewed (approved):
Looks good!

emilio-alvarez committed:
Merge branch 'master' into 17-view-stream-layout
